apps/src/pythonlab/patches/pythonScriptUtils.ts

```diff
@@ -23,3 +14,98 @@ export function applyPatches(originalCode: string) {
   }
   return finalCode;
 }
+
+// Pyodide uses the same interpreter for the lifetime of the browser tab.
+// In order to ensure we get updated user code for each run, we delete the
+// modules created by the user code from the sys.modules cache.
+export function deleteCachedUserModules(
+  source: MultiFileSource,
+  excludedFileName: string
+) {
+  let result = '\n';
+  for (const file of Object.values(source.files)) {
+    if (file.name !== excludedFileName) {
+      const filePath = getModuleName(file.id, source);
+      result += `
+if "${filePath}" in sys.modules:
+  del sys.modules['${filePath}']
+`;
+    }
+  }
+  return result;
+}
+
+// Write source to the Pyodide file system.
+// This enables python files to import from other files in the project.
+export function writeSource(
+  source: MultiFileSource,
+  currentFolderId: string,
+  currentPath: string,
+  pyodide: PyodideInterface
+) {
+  // Find all files in this folder and write them.
+  Object.values(source.files)
+    .filter(f => f.folderId === currentFolderId)
+    .forEach(file => {
+      pyodide.FS.writeFile(`${currentPath}${file.name}`, file.contents);
+    });
+  Object.values(source.folders)
+    .filter(f => f.parentId === currentFolderId)
+    .forEach(folder => {
+      // Create folder if it doesn't exist.
+      const newPath =
+        currentPath.length === 0
+          ? `${folder.name}`
+          : `${currentPath}${folder.name}`;
```

Comment:
Can't this just be `\`${currentPath}{folder.name}\`` w/o the ternary?

It's already assuming that `currentPath` is defined, and if the length is 0, it'd just output an empty string anyway, right?

Reply:
I think you're right, at one point I had a `/` in here but it got moved to somewhere else.